### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by ESaracay**
This code has all been reviewed but was merged with changes and not the master branch.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.lang.Thread;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+/**
+ * Runs when the form for a new comment is submitted. The comment
+ * is then stored permenantly with DatastoreService.
```
Comment (zmr) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java:18]: nitpick: permanently

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,12 +12,111 @@
...
+function startChat() {
+  grabComments();
+  canvas = document.getElementById('myCanvas');
+  context = canvas.getContext('2d');
+  canvas.width = window.innerWidth;
+  canvas.height = window.innerHeight;
+  init();
+}
+
+class Bubble {
```
Comment (zmr) [inline portfolio/src/main/webapp/script.js:32]: script.js contains logic for both animation and the chat interface. Given that the two are functionally unrelated, you might want to separate them into two separate js files.

**Comment by tylerhou** [comment]
This code has a lot of overlap with #9; do we still need this PR opened? Can we close in favor of #9?

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (tylerhou) [reply portfolio/src/main/webapp/script.js:32]: On the other hand, multiple JS files increase load times.

In an ideal world, we would have two JS files + a build systems that bundles the two files together into one, but IMO this is fine. (Not to say you shouldn't implement Zach's suggestion; that's your choice.)

**Pull request closed without merging**